main/src/core/MonoDevelop.Core/MonoDevelop.Core.Execution/RemoteProcessConnection.cs

```
@@ -430,7 +435,9 @@ void AbortConnection (string message = null, bool isAsync = false)
 		void AbortPendingMessages ()
 		{
 			lock (messageWaiters) {
-				foreach (var m in messageWaiters.Values)
+				// capture the original values because they can change while we're enumerating
+				var originalValues = messageWaiters.Values.ToArray ();
+				foreach (var m in originalValues)
```

If the collection can change, is it possible a value will be *added* while we're enumerating or is it only possible for them to be *removed*? If they can be added then we'd silently discard a value here because we clear the list at the end without checking for newly added items.